sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -0,0 +1,871 @@
+
+        _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+        RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_BEGIN_OBJECT);
+        _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+
+        while (ref_json_reader->token.kind != AZ_JSON_TOKEN_END_OBJECT)
+        {
+          RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_PROPERTY_NAME);
+
+          if (az_json_token_is_text_equal(
+                  &ref_json_reader->token,
+                  AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_FILE_NAME)))
+          {
+            _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+            RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_STRING);
+            update_manifest->files[files_index].file_name = ref_json_reader->token.slice;
+          }
+          else if (az_json_token_is_text_equal(
+                       &ref_json_reader->token,
+                       AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_SIZE_IN_BYTES)))
+          {
+            _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+            RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_NUMBER);
+
+            _az_RETURN_IF_FAILED(az_json_token_get_int64(
+                &ref_json_reader->token, &update_manifest->files[files_index].size_in_bytes));
+          }
+          else if (az_json_token_is_text_equal(
+                       &ref_json_reader->token,
+                       AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_HASHES)))
+          {
+            _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+            RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_BEGIN_OBJECT);
+            _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+
+            update_manifest->files[files_index].hashes_count = 0;
+
+            while (ref_json_reader->token.kind != AZ_JSON_TOKEN_END_OBJECT)
+            {
+              uint32_t hashes_count = update_manifest->files[files_index].hashes_count;
+
+              RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_PROPERTY_NAME);
+              update_manifest->files[files_index].hashes[hashes_count].hash_type
+                  = ref_json_reader->token.slice;
+              _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+              RETURN_IF_JSON_TOKEN_NOT_TYPE((ref_json_reader), AZ_JSON_TOKEN_STRING);
+              update_manifest->files[files_index].hashes[hashes_count].hash_value
+                  = ref_json_reader->token.slice;
+
+              update_manifest->files[files_index].hashes_count++;
+
+              _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
+            }
+          }
+          /*
+           * C SDK will not support delta updates at this time, so relatedFiles,
+           * downloadHandler, and mimeType are not exposed or processed.
```

Comment:
Should this be a remark in the API docs?

Reply:
Good point. I'll add this since it's documentation updates only.

Reply:
https://github.com/Azure/azure-sdk-for-c/pull/2421